### PR TITLE
Remove old string detection methods.

### LINKED ISSUE #3506: Make `rz_analysis_data_new_string()` and `rz_analysis_data_new()` use proper strings
See `librz/analysis/data.c` - it uses incomplete implementation of the string checking/search - it should share/reuse the code from `librz/util/str*.c`

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fix #3506

### CODE REVIEW

File: librz/analysis/data.c [hunk 1]
```diff
@@ -280,25 +251,40 @@ RZ_API void rz_analysis_data_free(RzAnalysisData *d) {
 	}
 }
 
-RZ_API RzAnalysisData *rz_analysis_data(RzAnalysis *analysis, ut64 addr, const ut8 *buf, int size, int wordsize) {
+/**
+ * @brief      { function_description }
```
Comment (XVilka) [inline librz/analysis/data.c]: Wrong syntax

File: librz/analysis/data.c [hunk 2]
```diff
@@ -280,25 +251,40 @@ RZ_API void rz_analysis_data_free(RzAnalysisData *d) {
...
 	if (size < 4) {
 		return NULL;
 	}
 	if (size >= word && is_invalid(buf, word)) {
-		return rz_analysis_data_new(addr, RZ_ANALYSIS_DATA_TYPE_INVALID, -1, buf, word);
+		return rz_analysis_data_new(addr, RZ_ANALYSIS_DATA_INFO_TYPE_INVALID, -1, buf, word);
 	}
 	{
-		int i, len = RZ_MIN(size, 64);
+		int len = RZ_MIN(size, 64);
```
Comment (XVilka) [inline librz/analysis/data.c]: Isn't `size_t` a better fit here? Or `ut64` if we want to support really huge strings

File: librz/core/canalysis.c [hunk 3]
```diff
@@ -2443,64 +2443,54 @@ RZ_API int rz_core_analysis_all(RzCore *core) {
 }
 
 RZ_API int rz_core_analysis_data(RzCore *core, ut64 addr, int count, int depth, int wordsize) {
```
Comment (XVilka) [inline librz/core/canalysis.c]: Please add Doxygen

**Review by XVilka — CHANGES_REQUESTED** [review]
Overall changes are good. Please fix the `rz-bindgen` build too:
```
Mismatched function annotation for rz_analysis_data at <rizin/librz/include/rz_analysis.h:1858:24> : None (was RZ_OWN at <rizin/librz/analysis/data.c:265:31>
Mismatched function annotation for rz_analysis_data at <rizin/librz/include/rz_analysis.h:1858:24> : None (was RZ_NONNULL at <rizin/librz/analysis/data.c:265:31>
```

File: librz/analysis/data.c (hunk 2 above)
Comment (wargio) [reply librz/analysis/data.c]: yes.

File: librz/analysis/data.c [hunk 4]
```diff
@@ -2,36 +2,26 @@
...
-		if (!IS_PRINTABLE(buf[i])) {
-			*len = i;
-			return 0;
-		}
+
+	RzUtilStrScanOptions opt = {
+		.buf_size = size,
+		.max_uni_blocks = 4,
+		.min_str_length = 4,
+		.prefer_big_endian = false,
```
Comment (XVilka) [inline librz/analysis/data.c]: Shouldn't we use endianness from the RzAnalysis here?

File: librz/core/core.c [hunk 5]
```diff
@@ -1749,35 +1749,24 @@ static void update_sdb(RzCore *core) {
...
-		if (!IS_PRINTABLE(buf[i])) {
-			*len = i;
-			return 0;
-		}
+
+	RzUtilStrScanOptions opt = {
+		.buf_size = size,
+		.max_uni_blocks = 4,
+		.min_str_length = 4,
+		.prefer_big_endian = false,
```
Comment (XVilka) [inline librz/core/core.c]: Same

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/analysis/data.c (hunk 4 above)
Comment (wargio) [reply librz/analysis/data.c]: hmmm probably

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
```
../librz/analysis/data.c
../librz/analysis/data.c:345:34: error: implicit conversion from enumeration type 'RzAnalysisDataInfoType' to different enumeration type 'RzAnalysisDataKind' [-Werror,-Wenum-conversion]
        rz_return_val_if_fail(a && buf, RZ_ANALYSIS_DATA_INFO_TYPE_INVALID);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../librz/include/rz_util/rz_assert.h:112:12: note: expanded from macro 'rz_return_val_if_fail'
                        return (val); \
                        ~~~~~~  ^~~
1 error generated.
```

**Review by XVilka — APPROVED** [review]

**Pull request merged**